app/lib/features/pins/widgets/fake_link_attachment_item.dart

```diff
@@ -0,0 +1,153 @@
+import 'package:logging/logging.dart';
+import 'package:skeletonizer/skeletonizer.dart';
+
+final _log = Logger('a3::pin::fake-pin-link-attachment-item');
+
+class FakeLinkAttachmentItem extends ConsumerWidget {
+  final String pinId;
+
+  const FakeLinkAttachmentItem({
+    super.key,
+    required this.pinId,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final pinData = ref.watch(pinProvider(pinId));
+    return pinData.when(
+      data: (pin) {
+        return fakeLinkAttachmentItemUI(
+          context,
+          ref,
+          pin,
+        );
+      },
+      loading: () => Skeletonizer(
+        child: Text(L10n.of(context).loadingPin),
+      ),
+      error: (err, st) {
+        _log.severe('Error loading pin', err, st);
+        return Text(
+          L10n.of(context).errorLoadingPin(err),
+        );
+      },
+    );
+  }
+
+  Widget fakeLinkAttachmentItemUI(
+    BuildContext context,
+    WidgetRef ref,
+    ActerPin pin,
+  ) {
+    String link = pin.url() ?? '';
+    if (link.isEmpty) return const SizedBox.shrink();
+    return Container(
+      margin: const EdgeInsets.symmetric(horizontal: 12),
+      decoration: BoxDecoration(
+        color: Theme.of(context).colorScheme.surface,
+        borderRadius: BorderRadius.circular(8),
+        border: Border.all(color: Theme.of(context).unselectedWidgetColor),
+      ),
+      child: ListTile(
+        leading: const Icon(Atlas.link),
+        onTap: () => openLink(link, context),
+        title: Text(
+          link,
+          maxLines: 2,
+          overflow: TextOverflow.ellipsis,
+          style: Theme.of(context).textTheme.labelMedium,
+        ),
+        trailing: optionMenu(ref, pin, link),
+      ),
+    );
+  }
+
+  Widget optionMenu(
+    WidgetRef ref,
+    ActerPin pin,
+    String link,
+  ) {
+    //Get my membership details
+    final membership =
+        ref.read(roomMembershipProvider(pin.roomIdStr())).valueOrNull;
+
+    return PopupMenuButton<String>(
+      key: const Key('fake-pink-link-menu-options'),
+      icon: const Icon(Icons.more_vert),
+      itemBuilder: (context) => [
+        //Check for can post pin permission
+        if (membership?.canString('CanPostPin') == true)
+          PopupMenuItem<String>(
+            key: const Key('fake-pink-link-edit'),
+            onTap: () {
+              showAddEditLinkBottomSheet(
+                context: context,
+                pinLink: link,
+                onSave: (title, newLink) async {
+                  if (link == newLink) Navigator.pop(context);
+                  await handleLinkBackwardSupportOnEdit(
+                    context,
+                    ref,
+                    pin,
+                    title,
+                    newLink,
+                  );
+                },
+              );
+            },
+            child: Text(L10n.of(context).edit),
+          ),
+        PopupMenuItem<String>(
```

Comment:
how can the user delete a pin if they don't have pin-edit-permissions? I think the entire menu needs to be behind that permission flag, no?